Sources/SwiftDocCUtilities/Action/Actions/PreviewAction.swift

```diff
@@ -275,6 +275,8 @@ extension DocumentationContext {
     func previewPaths() throws -> [String] {
         let urlGenerator = PresentationURLGenerator(context: self, baseURL: URL(string: "/")!)
         
+        let rootModules = try rootModules.filter { try !entity(with: $0).isVirtual }
```

Comment:
I think there might be other places in `DocumentationContext` where we want to apply the filtering. For example, I don't believe that `soleRootModuleReference` should consider virtual modules. It might be useful to expose a property like `nonVirtualRootModules` (is there a better word for non-virtual?) that we can use in more places.

Reply:
I updated the `soleRootModuleReference` for the time being.

Reply:
I take that back, filtering in `soleRootModuleReference` would break snippet-only packages, so it should stay there for now.

Reply:
Not updating `soleRootModuleReference` would cause wrong behavior for multi-language projects. For example, when building documentation for a Swift + Objective-C framework, the languages of article documents are inferred from `soleRootModuleReference`. When building a Swift + Objective-C framework that has snippets, there would actually be multiple modules and hence `soleRootModuleReference` would be nil and wouldn't be used to infer the languages of articles. 

This can probably be reproduced by adding snippets to `Tests/SwiftDocCTests/Test Bundles/MixedLanguageFramework.docc`, opening `/documentation/mixedlanguageframework/article` in the browser, and seeing that the language toggle at the top-right of the page would only show "Swift" rather than "Swift" and "Objective-C".

Reply:
I see, so we need different logic here. For now, I can address this by filtering out virtual modules, but allow a virtual module to be the sole root module reference if it is the only one.

It might be good to rethink or at least rename this property in the future.